### PR TITLE
Refactor CI pipelines, and add YAML NuGet package generation pipelines ( for CPU, MKLML, NoContribOps) 

### PR DESCRIPTION
**Pull request by jignparm**
Refactored the current build pipelines into templates so that common blocks can be reused.

Added new YAML pipelines for generating NuGet packages (including post-package tests). The 3 pipelines generate the CPU package, MKLML package and NoContribOps package.

### CODE REVIEW

File: tools/ci_build/github/azure-pipelines/linux-ci-pipeline.yml [hunk 1]
```diff
@@ -1,40 +1,8 @@
...
-
-    - script: 'tools/ci_build/github/linux/run_dockerbuild.sh -o ubuntu16.04 -d cpu -r $(Build.BinariesDirectory) -x "--use_mklml --use_tvm --build_wheel"'
-      displayName: 'Command Line Script'
-
-    - task: ms.vss-governance-buildtask.governance-build-task-component-detection.ComponentGovernanceComponentDetection@0
-      displayName: 'Component Detection'
-      condition: and(succeeded(), in(variables['Build.Reason'], 'IndividualCI', 'BatchedCI'))
-
-    - template: templates/clean-agent-build-directory-step.yml
+- template: templates/win-ci.yml
```
Comment (snnn) [inline tools/ci_build/github/azure-pipelines/linux-ci-pipeline.yml]: Maybe it need a better name? 
Why the Linux CI pipeline includes a template from win-ci.yml?

Comment (jignparm) [reply tools/ci_build/github/azure-pipelines/linux-ci-pipeline.yml]: Updated to "templates/linux-ci". Part of this PR was to test the all the individual CI pipelines at one time (the NuGet pipelines  were tested manually + passing).

File: tools/ci_build/github/azure-pipelines/bak_latest2/win-x86-nocontribops-ci-pipeline.yml [hunk 2]
```diff
@@ -0,0 +1,111 @@
+variables:
```
Comment (pranavsharma) [inline tools/ci_build/github/azure-pipelines/bak_latest2/win-x86-nocontribops-ci-pipeline.yml]: bak_latest2?

Comment (jignparm) [reply tools/ci_build/github/azure-pipelines/bak_latest2/win-x86-nocontribops-ci-pipeline.yml]: removed -- not used in final version

**Pull request merged**